distributed/tests/test_worker.py

```diff
@@ -1180,3 +1180,19 @@ def test_wait_for_outgoing(c, s, a, b):
     ratio = aa / bb
 
     assert 1 / 3 < ratio < 3
+
+
+@gen_cluster(client=True, ncores=[('127.0.0.1', 1)] * 20, timeout=30)
+def test_avoid_oversubscription(c, s, *workers):
+    np = pytest.importorskip('numpy')
+    x = c.submit(np.random.random, 1000000, workers=[workers[0].address])
+    yield wait(x)
+
+    with dask.config.set({'distributed.worker.max-connections': 1}):
+        futures = [c.submit(len, x, pure=False, workers=[w.address])
+                   for w in workers[1:]]
+
+        yield wait(futures)
+
+    assert len(workers[0].outgoing_transfer_log) < 18
+    assert sum(not not w.outgoing_transfer_log for w in workers) >= 3
```

Comment:
`not not` is weird. wouldn't `bool(w.outgoing_transfer_log)` work?

Reply:
or `sum(1 for w in workers if len(w.outgoing_transfer_log) > 0)` that might be even more explicit.

Reply:
Fixed to be more explicit.  I'm using `len(... for ... if ...)`